src/Slider/Slider.jsx

```diff
@@ -88,28 +97,102 @@ const Slider = class Slider extends React.Component {
     this.isDocumentScrolling = null;
   }
 
-  handleDocumentScroll() {
-    if (!this.props.touchEnabled) return;
-    this.isDocumentScrolling = true;
+  onDragStart(screenX, screenY, touchDrag, mouseDrag) {
+    window.cancelAnimationFrame.call(window, this.moveTimer);
+
+    if (this.props.orientation === 'vertical') {
+      this.originalOverflow = this.originalOverflow || document.documentElement.style.overflow;
+      document.documentElement.style.overflow = 'hidden';
+    }
+
+    this.setState({
+      isBeingTouchDragged: touchDrag,
+      isBeingMouseDragged: mouseDrag,
+      startX: screenX,
+      startY: screenY,
+    });
   }
 
-  handleOnTouchStart(ev) {
-    if (!this.props.touchEnabled) return;
+  onDragMove(screenX, screenY) {
+    window.cancelAnimationFrame.call(window, this.moveTimer);
+
+    this.moveTimer = window.requestAnimationFrame.call(window, () => {
+      this.setState({
+        deltaX: screenX - this.state.startX,
+        deltaY: screenY - this.state.startY,
+        mouseIsMoving: this.state.isBeingMouseDragged,
+      });
+    });
+  }
 
+  onDragEnd() {
     window.cancelAnimationFrame.call(window, this.moveTimer);
 
-    const touch = ev.targetTouches[0];
+    this.computeCurrentSlide();
+
     if (this.props.orientation === 'vertical') {
-      this.originalOverflow = this.originalOverflow || document.documentElement.style.overflow;
-      document.documentElement.style.overflow = 'hidden';
-      ev.preventDefault();
-      ev.stopPropagation();
+      document.documentElement.style.overflow = this.originalOverflow;
+      this.originalOverflow = null;
     }
+
     this.setState({
-      isBeingTouchDragged: true,
-      startX: touch.screenX,
-      startY: touch.screenY,
+      deltaX: 0,
+      deltaY: 0,
+      isBeingTouchDragged: false,
+      isBeingMouseDragged: false,
+      mouseIsMoving: false,
     });
+
+    this.isDocumentScrolling = this.props.lockOnWindowScroll ? false : null;
+  }
+
+  handleOnMouseDown(ev) {
+    if (!this.props.dragEnabled) return;
+
+    ev.preventDefault();
+    this.onDragStart(ev.screenX, ev.screenY, false, true);
+  }
+
+  handleOnMouseUp() {
+    if (!this.props.dragEnabled || !this.state.isBeingMouseDragged) return;
+
+    setTimeout(() => {
```

Comment:
Not a big fan of setTimeout in code.  Can you describe the problem you encountered that required the use of a timer?

Reply:
I totally understand, the reason i had to add it was to be able to handle clicking on slides that may contain a link but at the same time also not having clicks go through when you're dragging.

So to not have the state change immediately when you release the mouse button and set isBeingMouseDragged & mouseIsMoving to false which would let a click go through i had to delay it until the click has been registered.

Reply:
What about using `requestAnimationFrame`?

I am working on a component which allows dynamic content inside carousel, I used the method described in below blog post to calculate naturalWidth and naturalHeight, works great without magic setTimeouts.

https://stanko.github.io/react-rerender-in-component-did-mount/

Reply:
Any timer combined with setState is a bad idea.  The reason being that there's a chance that react might re-render and setState will reference an instance of the class that no longer exists.  You end up with a lot of warnings in the console.  handleOnMouseUp sets the timeout which calls onDragEnd which contains a setState.  Another thing about timers is that 100 might not be enough time if a user's computer is lagging for some reason.

Reply:
Let me think about this one for a bit.  I think there may be some css way to prevent the click event from bubbling.

Reply:
Yeah, it would definitely be best if we can have a solution without a timeout if possible.